### DISCUSSION TITLE
Change directory on the fly, while consult-*grep is active

### DISCUSSION THREAD

**Discussion by gamb** [minad/consult#1101] [Q&A] [answered]
When using `consult-grep` (etc.) it would often be handy to change the search directory without having to `C-g` to quit the minibuffer and then select a different directory in order to narrow results

I know it is possible to narrow with the `#` delimiter but this is still quite imprecise when I know e.g. I want to narrow to the current directory or one level above

A while ago I tried a hacky version using a trick to quit and then re-run the minibuffer: https://gist.github.com/gamb/8dbd6212d0147eae79b5628aace33058

**Comment by minad** [comment, marked as answer]
Afaik restarting the minibuffer in a different directory is the way to go. See also https://github.com/minad/consult/wiki#restart-consult-ripgrep-in-parent-directory.

**Reply by gamb** [reply]
Ah! Thank you @minad
